lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -114,45 +123,48 @@ def compact_messages(batch)
       # All messages are split into slices containing only unique keys, and
       # each slice is handles as its own batch.
       # @param messages [Array<Message>] List of messages.
-      # @return [void]
+      # @return [Array<ActiveRecord::Base>]
       def uncompacted_update(messages)
         BatchSlicer.
           slice(messages).
-          each(&method(:update_database))
+          each(&method(:update_database)).flatten
```

Comment:
Probably should use `map` instead of `each` here since we're using the return value.